src/actions/index.js

```diff
@@ -131,34 +131,20 @@ export function redo() {
     };
 }
 
+const UNDO = createAction('UNDO')();
 export function undo() {
-    return function(dispatch, getState) {
-        const history = getState().history;
-        dispatch(createAction('UNDO')());
-        const previous = history.past[history.past.length - 1];
-
-        // Update props
-        dispatch(
-            createAction('UNDO_PROP_CHANGE')({
-                itempath: getState().paths[previous.id],
-                props: previous.props,
-            })
-        );
-
-        // Notify observers
-        dispatch(
-            notifyObservers({
-                id: previous.id,
-                props: previous.props,
-            })
-        );
-    };
+    return undo_revert(UNDO);
```

Comment:
DRYing this section didn't change the functionality at all... but I discovered I could while researching another question: Can we remove the history from the store entirely when these buttons are hidden? The answer to that is no: in principle we could simplify it to saving just the last `past` entry and nothing else - which may be a worthwhile improvement for very large and long-running apps, it's possible the memory usage will eventually cause problems. But `revert` is still needed for error recovery.

Reply:
Could you open an issue to investigate/implement the improvement? Wonder how much of an issue this could be today with long-running apps updating graph/table components.

Reply:
> Could you open an issue to investigate/implement the improvement? Wonder how much of an issue this could be today with long-running apps updating graph/table components.

https://github.com/plotly/dash/issues/727